### PR TITLE
Optimizing native adapter

### PR DESCRIPTION
**Pull request by kofj**
R.T.

### CODE REVIEW

File: src/common/utils/registry/repository.go [hunk 1]
```diff
@@ -99,13 +101,6 @@ func (r *Repository) ListTag() ([]string, error) {
 		sort.Strings(tags)
 		tags = tagsResp.Tags
 
-		return tags, nil
-	} else if resp.StatusCode == http.StatusNotFound {
-
```
Comment (ywk253100) [inline src/common/utils/registry/repository.go]: The issue of distribution is not fixed yet, why we remove the workaround here?

File: src/replication/adapter/native/image_registry.go [hunk 2]
```diff
@@ -63,24 +62,21 @@ func (n native) FetchImages(filters []*model.Filter) ([]*model.Resource, error)
 		})
 	}
 
+	// replication execution should be failed if no any image found,
+	// and show the reason to user.
+	if resources == nil {
+		return nil, fmt.Errorf("Not find any image of name filter: %s, tag filter: %s", nameFilterPattern, tagFilterPattern)
```
Comment (ywk253100) [inline src/replication/adapter/native/image_registry.go]: Shouldn't return error here, if no resources returned, the execution will skip.

File: src/common/utils/registry/repository.go (hunk 1 above)
Comment (kofj) [reply src/common/utils/registry/repository.go]: This workaround will hide error message when it is true. Can we 
just hidden error in the UI to fix that's issue of harbor.

File: src/replication/adapter/native/image_registry.go (hunk 2 above)
Comment (kofj) [reply src/replication/adapter/native/image_registry.go]: But the execution status will become success if nothing to do. User maybe not know reason that's execution is success but task list is empty.

Comment (ywk253100) [reply src/replication/adapter/native/image_registry.go]: The status text will be filled with message : there is no resources to replicate.

File: src/common/utils/registry/repository.go (hunk 1 above)
Comment (ywk253100) [reply src/common/utils/registry/repository.go]: No. The method is used by other functions and it's a bug of registry API, we should not expose it to upper callers

Comment (kofj) [reply src/common/utils/registry/repository.go]: 👌, I have reverted.

**Pull request merged**